### PR TITLE
setup jest-puppeteer for UI test

### PR DESCRIPTION
**Pull request by Shulammite-Aso**
Has also created this at bevacqua/woofmark here bevacqua#73

### CODE REVIEW

**Comment by Shulammite-Aso** [comment]
Hi @jywarren @NitinBhasneria i think this works fine now.


tagging @shreyaa-sharmaa and @keshav234156 too. followed from here https://github.com/jywarren/woofmark/pull/7


Thanks!!

**Comment by jywarren** [comment]
This looks really great! Thank you!

File: test/ui-tests/title.test.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const fs = require('fs');
+beforeAll(async () => {
+  path = fs.realpathSync('file://../index.html');
+  await page.goto('file://' + path, {waitUntil: 'domcontentloaded'});
+});
+
+describe('Title of the page', () => {
+  test('Title of the page', async () => {
+    const title = await page.title();
+    expect(title).toBe('woofmark');
```
Comment (jywarren) [inline test/ui-tests/title.test.js:11]: What would be some next steps in terms of basic tests we could add? Question for everybody! cc @NitinBhasneria @shreyaa-sharmaa @keshav234156 🎉

**Review by jywarren — APPROVED** [review]

File: test/ui-tests/title.test.js (hunk 1 above)
Comment (NitinBhasneria) [reply test/ui-tests/title.test.js:11]: I will be adding the test for #2 and #3

**Comment by cypherean** [comment]
I'll add the test for #5 here.
I think we'll have to plan the tests that we'll have here and the ones we have in PLE in regards to formatting. With every change that we'll have in here, we need to describe tests in PLE that'll make sure that none of the existing functions break and check that the change meets our requirements/ acts as was intended here.

**Comment by cypherean** [comment]
Hey @Shulammite-Aso this is not the same as #7 . The test you integrated there are merged into master where as this is merged in `plots2`. This PR doesn't integrate the tests into `plots2`. The only reason travis passed was because it couldn't find any tests to run at all since there are no included dependency of jest, puppeteer or jest-puppeteer in this one. Attaching screenshot of travis test:
Screenshot from 2020-07-14 16-04-16


The downside of this is that all the fixes and feature additions we make are merged into `plots2` branch but the tests are only conducted on `master` branch which doesn't incorporate these changes, hence we won't be able to define our tests here. Please open a new PR with updated `package.json`, adding the required dependency and update `travis.yml` to include the test script. Similar to bevacqua#73. 

Also, you might have to switch `puppeteer` to an older version. This is the output I receive while running `npm install` and I remember receiving the same when working on the Editor. Though we haven't faced any errors for the same, but let's be on the safe side so that we don't encounter any trouble in the future. 
Screenshot from 2020-07-14 16-06-08

**Comment by Shulammite-Aso** [comment]
> Hey @Shulammite-Aso this is not the same as #7 . The test you integrated there are merged into master where as this is merged in plots2. This PR doesn't integrate the tests into plots2. The only reason travis passed was because it couldn't find any tests to run at all since there are no included dependency of jest, puppeteer or jest-puppeteer in this one. Attaching screenshot of travis test:

Yes, have seen this already. Sorry about it.  The dependencies would be included in the next test i'll be pushing, with the test script in `travis.yml`.

**Comment by NitinBhasneria** [comment]
I think @Shulammite-Aso you have got some small error in `npm run scripts`. Please fix this.
Screenshot from 2020-07-14 22-23-56
